### PR TITLE
Update/vault v1.1.0

### PR DESCRIPTION
**Pull request by danoctavian**

### CODE REVIEW

File: src/KernelStrategy.sol [hunk 1]
```diff
@@ -259,22 +216,12 @@ contract KernelStrategy is Vault {
...
-        SafeERC20.safeTransferFrom(IERC20(asset_), caller, address(this), assets);
-        _mint(receiver, shares);
+        super._deposit(asset_, caller, receiver, assets, shares, baseAssets);
 
         StrategyStorage storage strategyStorage = _getStrategyStorage();
         if (strategyStorage.syncDeposit) {
             _stake(asset_, assets, IStakerGateway(strategyStorage.stakerGateway));
         }
-
-        emit DepositAsset(caller, receiver, asset_, assets, shares);
```
Comment (danoctavian) [inline src/KernelStrategy.sol:277]: created a version of this for BaseVault. This should have existed

File: src/KernelStrategy.sol [hunk 2]
```diff
@@ -59,44 +54,6 @@ contract KernelStrategy is Vault {
...
-    /**
-     * @notice Initializes the vault.
-     * @param admin The address of the admin.
-     * @param name The name of the vault.
-     * @param symbol The symbol of the vault.
-     * @param decimals The decimals of the vault.
-     * @param baseWithdrawalFee The base withdrawal fee.
-     * @param countNativeAsset Whether to count the native asset.
-     */
-    function initialize(
```
Comment (danoctavian) [inline src/KernelStrategy.sol:71]: decided to delete this, we only have it for hasAllocators, so why introduce more logic

File: src/KernelStrategy.sol [hunk 3]
```diff
@@ -329,8 +276,8 @@ contract KernelStrategy is Vault {
             revert AssetNotActive();
         }
 
-        VaultStorage storage vaultStorage = _getVaultStorage();
-        vaultStorage.totalAssets -= _convertAssetToBase(asset_, assets);
+        _subTotalAssets(_convertAssetToBase(asset_, assets));
```
Comment (danoctavian) [inline src/KernelStrategy.sol:279]: if you look into BaseVault this is conditionally altered now

File: src/KernelStrategy.sol [hunk 4]
```diff
@@ -259,22 +216,12 @@ contract KernelStrategy is Vault {
...
-        if (!_getAssetStorage().assets[asset_].active) {
-            revert AssetNotActive();
-        }
-
-        VaultStorage storage vaultStorage = _getVaultStorage();
-        vaultStorage.totalAssets += baseAssets;
-
-        SafeERC20.safeTransferFrom(IERC20(asset_), caller, address(this), assets);
-        _mint(receiver, shares);
+        super._deposit(asset_, caller, receiver, assets, shares, baseAssets);
```
Comment (danoctavian) [inline src/KernelStrategy.sol:219]: exact same logic runs in deposit

**Review by dan13ram — APPROVED** [review]

**Review by MrDeadCe11 — APPROVED** [review]
only comment I have is we should probably update the deploy scripts to set hasAllocator when necessary.

File: script/bnb/deploy_ynwbnbk.s.sol [hunk 5]
```diff
@@ -56,9 +56,16 @@ contract DeployYnWBNBkStrategy is BaseScript {
         uint8 decimals = 18;
         uint64 baseWithdrawalFee = 0;
         bool countNativeAsset = true;
-
+        bool alwaysComputeTotalAssets = true;
```
[the remark this answers is not in the document]
Comment (danoctavian) [reply script/bnb/deploy_ynwbnbk.s.sol:59]: yeah good points

**Pull request merged**